src/activities/ActivityUsageEntity.js

```diff
@@ -86,41 +86,95 @@ export class ActivityUsageEntity extends Entity {
 		activityCollectionHref && this._subEntity(ActivityUsageCollectionEntity, activityCollectionHref, onChange);
 	}
 
-	_getDueDateSubEntity() {
-		return this._entity
-			&& this._entity.getSubEntityByClass(Classes.dates.dueDate);
-	}
-
 	/**
 	 * @returns {string} Due date of the activity usage
 	 */
 	dueDate() {
-		const dueDate = this._getDueDateSubEntity();
+		const dueDate = this._getDateSubEntity(Classes.dates.dueDate);
 		return dueDate && dueDate.properties.date;
 	}
 
 	/**
 	 * @returns {bool} Whether or not the edit due date action is present on the activity usage entity
 	 */
 	canEditDueDate() {
-		const dueDate = this._getDueDateSubEntity();
-		return (dueDate && dueDate.hasActionByName(Actions.activities.update))
-			|| this._entity.hasActionByName(Actions.activities.startAddNew);
+		return this._canEditDate(Classes.dates.dueDate);
 	}
 
 	/**
  	 * Updates the due date of the activity usage entity to the date specified
-	 * @param {string} dateString Date string to set as the due date, or empty string to clear the due date
+	 * @param {string} dateValue Date string to set as the due date, or empty string to clear the due date
+	 */
+	async setDueDate(dateValue) {
+		await this._setDate(dateValue, Classes.dates.startDate, 'dueDate');
```

Comment:
Should the 2nd param be `Classes.dates.dueDate`?

Reply:
o dang, so it should - your eyes are better than mine!

Reply:
It sure should be!